config/channel/resources/kafkachannel-crd.yaml

```diff
@@ -31,12 +31,197 @@ spec:
         status: { }
       schema:
         openAPIV3Schema:
+          description: 'KafkaChannel is a resource representing a Channel that is backed by a topic of an Apache Kafka cluster.'
           type: object
-          # this is a work around so we don't need to flush out the
-          # schema for each version at this time
-          #
-          # see issue: https://github.com/knative/serving/issues/912
-          x-kubernetes-preserve-unknown-fields: true
+          properties:
+            spec:
+              description: Spec defines the desired state of the Channel.
+              type: object
+              properties:
+                numPartitions:
+                  description: NumPartitions is the number of partitions of a Kafka topic. By default, it is set to 1.
```

Comment:
The `default` field is GA since 1.17. Maybe it makes sense to add it here for extra documentation purpose?